procServUtils/manage.py

```diff
@@ -226,6 +248,46 @@ def delproc(conf, args):
 
     #sys.stdout.write("# systemctl stop procserv-%s.service\n"%args.name)
 
+def renameproc(conf, args):
+    check_req(conf, args)
+    from .generator import run
+
+    # copy settings from previous conf
+    items = conf.items(args.name)
+    new_conf = ConfigParser()
+    new_conf.add_section(args.new_name)
+    for item in items:
+        new_conf.set(args.new_name, item[0], item[1])
+
+    # create new conf file with old settings
+    addconf(args.new_name, new_conf, args.user, args.force)
+
+    # delete previous proc
+    args.force = True
+    delproc(conf, args)
+
+    # generate service files
+    outdir = getgendir(args.user)
+    run(outdir, user=args.user)
+
+    # register systemd service
+    argusersys = '--user' if args.user else '--system'
+    SP.check_call([systemctl,
+                   argusersys,
+                   'enable',
+                   "%s/procserv-%s.service"%(outdir, args.new_name)])
+
+    _log.info('Trigger systemd reload')
+    SP.check_call([systemctl,
+                   argusersys,
+                   'daemon-reload'], shell=False)
```

Comment:
I'm not sure about the ordering here, but I think the `daemon-reload` should come first.  As I understand it, a reload is needed after manually editing any systemd config files, eg. before any new units are recognised.

I'm also wondering it an explicit `disable` is needed for the original service name?  Have you verified that the symlinks are being cleaned up?  (eg. `ls -l /etc/systemd/system/multi-user.target.wants/`)

Reply:
I agree with @mdavidsaver, the order should be:
1. stop and disable old unit (calling disable is required if we don't want to leave any mess behind)
2. modify unit files
3. daemon-reload
4. enable and start new unit

Reply:
Ok, good catch! I copied this logic from the ``addproc`` function, so I will fix the order also on that function.